### PR TITLE
Postgresql upsert bug

### PR DESCRIPTION
**Pull request by epuaphios**
Postgresql upsert  bug

Screenshot from 2023-04-05 16-16-51

### CODE REVIEW

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/TriggerRelationMapper.xml [hunk 1]
```diff
@@ -50,6 +50,8 @@
     <insert id="upsert">
         INSERT INTO t_ds_trigger_relation (trigger_code, trigger_type, job_id, create_time, update_time) VALUES(
             #{triggerRelation.triggerCode},#{triggerRelation.triggerType},#{triggerRelation.jobId},#{triggerRelation.createTime},#{triggerRelation.updateTime})
-        ON DUPLICATE KEY UPDATE update_time = #{triggerRelation.updateTime};
+        ON CONFLICT (trigger_code, trigger_type, job_id)
```
Comment (ruanwenjun) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/TriggerRelationMapper.xml]: Could you please split this method to insert/update?

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: deploy/docker/docker-compose.yml [hunk 2]
```diff
@@ -44,6 +44,8 @@ services:
       ZOO_4LW_COMMANDS_WHITELIST: srvr,ruok,wchs,cons
     volumes:
       - dolphinscheduler-zookeeper:/bitnami/zookeeper
+    ports:
+      - "2181:2181"
```
Comment (zhongjiajie) [inline deploy/docker/docker-compose.yml]: export no public ports is not the best practice

File: dolphinscheduler-api/src/main/docker/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#
+#hamileler d vitamini kullanabilir mi
```
Comment (zhongjiajie) [inline dolphinscheduler-api/src/main/docker/Dockerfile]:
```suggestion
#
```

**Review by zhongjiajie — CHANGES_REQUESTED** [review]

File: deploy/docker/docker-compose.yml (hunk 2 above)
Comment (epuaphios) [reply deploy/docker/docker-compose.yml]: yes but i running local test application not found zookeeper and postgresql

File: dolphinscheduler-api/src/main/docker/Dockerfile (hunk 3 above)
Comment (epuaphios) [reply dolphinscheduler-api/src/main/docker/Dockerfile]: sorry my fault

File: deploy/docker/docker-compose.yml [hunk 4]
```diff
@@ -110,6 +108,8 @@ services:
     image: ${HUB}/dolphinscheduler-master:${TAG}
     profiles: ["all"]
     env_file: .env
+    ports:
+      - "5679:5679"
```
Comment (SbloodyS) [inline deploy/docker/docker-compose.yml]: Please remove this.

File: deploy/docker/docker-compose.yml [hunk 5]
```diff
@@ -128,6 +128,8 @@ services:
     image: ${HUB}/dolphinscheduler-worker:${TAG}
     profiles: ["all"]
     env_file: .env
+    ports:
+      - "1235:1235"
```
Comment (SbloodyS) [inline deploy/docker/docker-compose.yml]: Same here.

**Review by SbloodyS — CHANGES_REQUESTED** [review]

File: deploy/docker/docker-compose.yml (hunk 4 above)
Comment (SbloodyS) [reply deploy/docker/docker-compose.yml]: Sorry. My mistake. Reverted this.

**Comment by SbloodyS** [comment]
Please resolve conflicts. @epuaphios

**Comment by Alixxxxxxx** [comment]
3.2.2 this bug is also in.

**Pull request closed without merging**